x/wasm/keeper/msg_dispatcher.go

```diff
@@ -153,6 +154,24 @@ func (d MessageDispatcher) DispatchSubmessages(ctx sdk.Context, contractAddr sdk
 	return rsp, nil
 }
 
+func filterEvents(events ...[]sdk.Event) []sdk.Event {
+	// pre-allocate space for efficiency
+	cap := 0
+	for _, evts := range events {
+		cap += len(evts)
+	}
```

Comment:
🤔  You can reduce complexity of this method a lot by just accepting `[]sdk.Event`. On the caller side just do: `filterEvents(append(em.Events(), events...))`. I would not expect thousands of events that require some optimization.

Reply:
Okay, in Rust everytime you heap allocate to make vectors (slices), it is considered "expensive". I guess we do not micro-optimize so aggressively in Go.

I will simplify